internal/controllers/gateway/gateway_utils_test.go

```diff
@@ -130,15 +130,25 @@ func TestGetListenerStatus_no_duplicated_Detached_condition(t *testing.T) {
 	require.NoError(t, err)
 	require.Len(t, statuses, 1, "only one listener status expected as only one listener was defined")
 	listenerStatus := statuses[0]
-	assertOnlyOneConditionOfType(t, listenerStatus.Conditions, gatewayv1beta1.ListenerConditionAccepted)
+	assertOnlyOneConditionOfTypes(t, listenerStatus.Conditions,
+		gatewayv1beta1.ListenerConditionAccepted,
+		gatewayv1beta1.ListenerConditionReady,
+		gatewayv1beta1.ListenerConditionProgrammed,
+		gatewayv1beta1.ListenerConditionResolvedRefs,
+		gatewayv1beta1.ListenerConditionConflicted,
+	)
```

Comment:
Does this mean that other conditions can be duplicated? 🤔 

How about just checking `listenerStatus.Conditions` for duplicates. It should do the same thing, right?